examples/arduino-uno/src/bin/uno-pin-change-interrupt.rs

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Example of enabling and handling pin change interrupts
+ *
+ * In this example we can get an interrupt when pin 2 changes
+ * and use that to move a stepper motor.
+ *
+ */
+#![no_std]
+#![no_main]
+#![feature(abi_avr_interrupt)]
+
+use arduino_hal;
+use panic_halt as _;
+
+use core:: sync::atomic::{AtomicBool, Ordering};
+
+static PIN_CHANGED: AtomicBool = AtomicBool::new(false);
+
+//This function is called on change of pin 2
+#[avr_device::interrupt(atmega328p)]
+#[allow(non_snake_case)]
+fn PCINT2() {
+	avr_device::interrupt::free(|_cs| {
+        PIN_CHANGED.store(true, Ordering::SeqCst);
+    });
```

Comment:
There is no need to have this critical section here.  For one, interrupts are already disabled when entering an ISR.  And secondly, as this variable is "atomic", no external synchronization is necessary to ensure safe mutation.

```suggestion
    PIN_CHANGED.store(true, Ordering::SeqCst);
```

(atomic really means "emulated atomic" here as AVR does not really have any atomic operations)